### PR TITLE
05 images

### PR DESCRIPTION
**Pull request by dykyys**
completed home work 05-images without additional task

### CODE REVIEW

File: api/auth/auth.controller.js [hunk 1]
```diff
@@ -1,19 +1,25 @@
...
 const { createVerificationToken } = require('../../services/token.service');
+const { avatar } = require('../../services/avatar.services');
 
 const registrationContoller = async (req, res, next) => {
   try {
     const { body } = req;
-    console.log(body);
+    const variant = 'female';
+    const image = await avatar.generate(`${body.email}`, variant);
+    image.png().toFile(`public/images/${body.email}.png`);
```
Comment (odykyi) [inline api/auth/auth.controller.js]: move `public/images/${body.email}.png` to config 


export const yourFunction = (id) => `public/images/${id}.png`

File: api/auth/auth.controller.js [hunk 2]
```diff
@@ -1,19 +1,25 @@
...
+
     const hachedPassword = await bcrypt.hash(body.password, +process.env.SALT);
     const newUser = await UserDB.createUser({
       ...body,
+      avatarURL: `http:3000/images/${body.email}.png`,
```
Comment (odykyi) [inline api/auth/auth.controller.js]: move protocol, host, port , File Fomat to config

File: api/auth/auth.controller.js [hunk 3]
```diff
@@ -136,11 +142,41 @@ const renewalSubContoller = async (req, res, next) => {
...
+      return;
+    }
+
+    if (!file) {
+      res.status(400).json({
+        message: `Avatar not found`,
+      });
+      return;
+    }
+    const avatarURL = `http://localhost:3000/images/${file.filename}`;
```
Comment (odykyi) [inline api/auth/auth.controller.js]: move to config

File: index.js [hunk 4]
```diff
@@ -19,12 +21,18 @@ const runServer = async () => {
 
     const app = express();
 
+    app.use('/', express.static(path.resolve(__dirname, 'public')));
+
     app.use(express.json());
     app.use(cors({ origin: 'http://localhost:3000' }));
```
Comment (odykyi) [inline index.js]: move url to config

File: index.js [hunk 5]
```diff
@@ -19,12 +21,18 @@ const runServer = async () => {
...
 
     app.use('/contacts', contactsRouter);
     app.use('/auth', authRouter);
     app.use('/users', authRouter);
+    app.use(
+      '/images',
+      express.static(path.resolve(__dirname, 'public/images')),
```
Comment (odykyi) [inline index.js]: move path to config

File: middlewares/fileUploader.middleware.js [hunk 6]
```diff
@@ -0,0 +1,24 @@
+const multer = require('multer');
+
+const avatarUploader = () => {
+  const storage = multer.diskStorage({
+    destination: function (req, file, cb) {
+      cb(null, 'public/images');
```
Comment (odykyi) [inline middlewares/fileUploader.middleware.js]: move path to config

File: services/avatar.services.js [hunk 7]
```diff
@@ -0,0 +1,10 @@
+const path = require('path');
+const AvatarGenerator = require('avatar-generator');
+const avatar = new AvatarGenerator({
+  parts: ['background', 'face', 'clothes', 'head', 'hair', 'eye', 'mouth'],
+
+  partsLocation: path.join(__dirname, '../img'),
```
Comment (odykyi) [inline services/avatar.services.js]: move partsLocation to config

**Review by odykyi — CHANGES_REQUESTED** [review]
comments

**Comment by dykyys** [comment]
fix problems

File: api/auth/auth.controller.js [hunk 8]
```diff
@@ -1,19 +1,32 @@
...
     const { body } = req;
-    console.log(body);
+
     const hachedPassword = await bcrypt.hash(body.password, +process.env.SALT);
     const newUser = await UserDB.createUser({
       ...body,
       password: hachedPassword,
     });
+    const currenUserById = await UserDB.findUserById(newUser._id);
+    madeAvatar(newUser._id);
```
Comment (odykyi) [inline api/auth/auth.controller.js]: await

**Comment by dykyys** [comment]
del images, add await madeAvatar

**Review by odykyi — APPROVED** [review]

**Pull request merged**